data-prepper-plugins/opensearch/src/test/java/com/amazon/dataprepper/plugins/sink/opensearch/index/TraceAnalyticsRawIndexManagerTests.java

```diff
@@ -0,0 +1,224 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.opensearch.OpenSearchException;
+import org.opensearch.action.admin.cluster.settings.ClusterGetSettingsRequest;
+import org.opensearch.action.admin.cluster.settings.ClusterGetSettingsResponse;
+import org.opensearch.action.admin.indices.alias.get.GetAliasesRequest;
+import org.opensearch.client.ClusterClient;
+import org.opensearch.client.IndicesClient;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.ResponseException;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestHighLevelClient;
+import org.opensearch.client.indices.CreateIndexRequest;
+import org.opensearch.client.indices.GetIndexRequest;
+import org.opensearch.client.indices.GetIndexTemplatesRequest;
+import org.opensearch.client.indices.GetIndexTemplatesResponse;
+import org.opensearch.client.indices.PutIndexTemplateRequest;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class TraceAnalyticsRawIndexManagerTests {
+    private static final String INDEX_ALIAS = "trace-raw-index-alias";
+
+    private TraceAnalyticsRawIndexManager traceAnalyticsRawIndexManager;
+
+    @Mock
+    private RestHighLevelClient restHighLevelClient;
+
+    @Mock
+    private OpenSearchSinkConfiguration openSearchSinkConfiguration;
+
+    @Mock
+    private ClusterClient cluster;
+
+    @Mock
+    private ClusterGetSettingsResponse clusterGetSettingsResponse;
+
+    @Mock
+    private IndexConfiguration indexConfiguration;
+
+    @Mock
+    private IndicesClient indicesClient;
+
+    @Mock
+    private GetIndexTemplatesResponse getIndexTemplatesResponse;
+
+    @Mock
+    private RestClient restClient;
+
+    @Mock
+    private ResponseException responseException;
+
+    @Before
+    public void setup() throws IOException {
+        initMocks(this);
+        traceAnalyticsRawIndexManager = new TraceAnalyticsRawIndexManager(restHighLevelClient, openSearchSinkConfiguration);
+
+        when(restHighLevelClient.cluster()).thenReturn(cluster);
+        when(cluster.getSettings(any(ClusterGetSettingsRequest.class), any(RequestOptions.class)))
+                .thenReturn(clusterGetSettingsResponse);
+
+        when(openSearchSinkConfiguration.getIndexConfiguration()).thenReturn(indexConfiguration);
+        when(indexConfiguration.getIndexAlias()).thenReturn(INDEX_ALIAS);
+        when(restHighLevelClient.indices()).thenReturn(indicesClient);
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
Please update this test to JUnit 5 and use `assertThrows` rather than `expected`.